src/common/classes/tree.h

```diff
@@ -106,24 +93,24 @@ enum LocType { locEqual, locLess, locGreat, locGreatEqual, locLessEqual };
 // an indexed dynamic array without increase of algorithm calculation costs (this is one
 // more classical B+ tree feature). This is also not done to improve tree performance a little
 //
-template <typename Value, typename Key = Value, typename Allocator = MallocAllocator,
+template <typename Value, typename Key = Value, typename Allocator = Firebird::MemoryPool,
 	typename KeyOfValue = DefaultKeyValue<Value>,
 	typename Cmp = DefaultComparator<Key> >
 class BePlusTree
 {
-	static const FB_SIZE_T LeafCount = LEAF_PAGE_SIZE / sizeof(Value);
-	static const FB_SIZE_T NodeCount = NODE_PAGE_SIZE / sizeof(void*);
+	static const FB_SIZE_T LEAF_COUNT = LEAF_PAGE_SIZE / sizeof(Value);
+	static const FB_SIZE_T NODE_COUNT = NODE_PAGE_SIZE / sizeof(void*);
 public:
 	explicit BePlusTree(Allocator *_pool)
-		: pool(_pool), level(0), root(NULL), defaultAccessor(this)
+		: pool(_pool), level(0), defaultAccessor(this)
```

Comment:
Also suggest to use reference (not pointer) to MemoryPool in all c-tors - like in the rest of our classes.

Reply:
Also OK

Reply:
Done